caravel/assets/javascripts/SqlLab/reducers.js

```diff
@@ -13,13 +13,39 @@ const defaultQueryEditor = {
 
 export const initialState = {
   alerts: [],
-  queries: [],
+  queries: {},
   queryEditors: [defaultQueryEditor],
   tabHistory: [defaultQueryEditor.id],
   tables: [],
   workspaceQueries: [],
 };
 
+function removeFromDict(state, arrKey, obj, idKey = 'id') {
```

Comment:
@ascott you have a good point. 
I am trying to stay with the style of the existing code.
Those functions are aimed to be reusable for dictionary/object modifications
I will rename them to the the Object suffix.